Src/Plugins/Security/Security.SharedStrings.txt

```diff
@@ -24,13 +24,13 @@
   $AdoPat=(^|[^2-7a-z])(?<refine>[2-7a-z]{52})([^2-7a-z]|$)
   $AwsCredentials=(?s)\b(?<id>(A3T[0-9A-Z]|AKIA|AGPA|AROA|AIPA|ANPA|ANVA|ASIA)[0-9A-Z]{16}).{1,500}?\b(?i)(?<key>[0-9a-z\/+]{40})
   $FacebookAppCredentials=(?si)(((facebook|fb)(.{0,20})?[`'"\s>](?<id>[0-9]{13,17})[^0-9].{1,500}?)|((facebook|fb)(.{0,20})?[`'"\s>](?<key>[0-9a-f]{32})([^0-9a-f]?)).{1,500}?){2}
-  $GitHubAppCredentials=(?s)\b(?<id>Iv1\.[0-9a-z]{16}).{1,500}?\b(?i)(?<key>[0-9a-z]{35,40})([^0-9a-f]|$)
+  $GitHubAppCredentials=(?s)((\b(?<id>Iv1\.[0-9a-z]{16}).{1,50}?)|(\b(?i)(?<key>[0-9a-z]{35,40})\b.{0,50}?)){2}
   $GitHubPat=(?si)(?<refine>github.{0,50}[^0-9a-z][0-9a-z]{40})([^0-9a-z]|$)
   $GoogleApiKey=\b(?<refine>AIza(?i)[0-9a-z-_]{35})([^0-9a-z-_]|$)
   $HockeyAppToken=(?si)hockey.{0,50}[`'"\s>]?(?<refine>[0-9a-f]{32})([^0-9a-f]|$)
   $MySqlConnectionStringAdo=(?i)(Port\s*=\s*([0-9]{4,5}).*)?(((Server\s*=\s*(?<host>[\w\-.]{3,90}))|(Uid=(?-i)(?<account>[a-z\@\-]{1,120})(?i))|(Pwd=(?<password>[^;]{8,128}))).*?){3}(.*Port\s*=\s*([0-9]{4,5}))?
-  $SqlConnectionStringAdo=(?i)(((server|data source)\s*=\s*[^;]{0,50}|(initial catalog|database)\s*=\s*[^;]{0,50}|(user id|uid)\s*=\s*[^;]{0,50}|(password|pwd)\s*=\s*[^;]{0,50}).+?){4}
-  $SqlConnectionStringJdbc=(?i)jdbc:sqlserver:\/\/(?<host>[^;]+);database\s*=\s*(?<database>[^;]+);user\s*=\s*(?<account>[^@]+)@[^;]+;password\s*=\s*(?<password>[^;]+);
+  $SqlConnectionStringAdo=(?i)(((server|data source)\s*=\s*(?<host>[^;]{0,50})|(initial catalog|database)\s*=\s*(?<database>[^;]{0,50})|(user id|uid)\s*=\s*(?<account>[^;]{0,50})|(password|pwd)\s*=\s*(?<password>[^;]{0,50})).+?){4}
```

Comment:
we have a problem with the following:
        [global::System.Configuration.DefaultSettingValueAttribute("Data Source=database;Initial Catalog=catalog;User ID=user;Password=password")]
        public string ELSConnectionString {

insteada of using [^;] as a way to find the  end, we can use \b. The below regex worked for the example:
(?i)(((server|data source)\s*=\s*(?<host>[^;]{0,50})|(initial catalog|database)\s*=\s*(?<database>[^;]{0,50})|(user id|uid)\s*=\s*(?<account>[^;]{0,50})|(password|pwd)\s*=\s*(?<password>.{0,50})\b).+?){4}

Reply:
@michaelcfanning ,what would be the best separator? because sometimes we are  getting wrong terminators...

---
In reply to: [573345752](https://github.com/microsoft/sarif-pattern-matcher/pull/183#discussion_r573345752) [](ancestors = 573345752)

Reply:
Are you sure that \b won't artificially truncate a password that contains something like an apostrophe?

password=L'tme{nN0w!;

---
In reply to: [573345752](https://github.com/microsoft/sarif-pattern-matcher/pull/183#discussion_r573345752) [](ancestors = 573345752)

Reply:
probably. Do u have any suggestion here?

---
In reply to: [573361827](https://github.com/microsoft/sarif-pattern-matcher/pull/183#discussion_r573361827) [](ancestors = 573361827,573345752)

Reply:
@eddynaka, it's super tricky! you can't properly identify a terminator unless you understand the source file the pattern is expressed in. also, our naive regex won't handle an escaped password properly, when rendered as a C# string.

in general, we'd need an understanding of how string literals are handled given the source file. and the escaping conventions for it. that's a big ask!

SQL connection strings themselves allow unicode-escaping, Raul told us. So there's another hard problem. 

My suggestion, try using whatever your good sense tells you gets best results for now, and we'll think hard about it later. How about this: pursue a greedy match that requires a semicolon, apostrophe or double-quote mark or space as the terminator?

password=2352ah28;" // password is 2352ah28;
password=2352ah28"; // password is 2352ah28"



---
In reply to: [573361577](https://github.com/microsoft/sarif-pattern-matcher/pull/183#discussion_r573361577) [](ancestors = 573361577,573345752)

Reply:
@eddynaka  had a list of delimiters that I've included in all properties of that string.  Also added @michaelcfanning's example to ensure it was correctly caught.

---
In reply to: [573363399](https://github.com/microsoft/sarif-pattern-matcher/pull/183#discussion_r573363399) [](ancestors = 573363399,573361827,573345752)